src/simple_model.jl

```diff
@@ -0,0 +1,153 @@
+export SimpleNLPModel, obj, grad, grad!, cons, cons!, jac_coord, jac, jprod,
+       jprod!, jtprod, jtprod!, hess, hprod, hprod!
+
+type SimpleNLPModel <: AbstractNLPModel
+  meta :: NLPModelMeta
+
+  counters :: Counters
+
+  # Functions
+  f :: Function
+  g :: Function
+  g! :: Function
+  H :: Function
+  c :: Function
+  c! :: Function
+  J :: Function
+end
+
+function SimpleNLPModel(x0; y0 = [], lvar = [], uvar = [], lcon = [], ucon = [],
+    nnzj = 0, nnzh = 0, obj = error, grad = error, grad! = error, hess =
```

Comment:
Ok, there is a difference. On the first `f == NI`  is `true`, but not on the second. One way to circunvent is to `try f(x)` and catch the exception. Since `SimpleNLPModel` is not about efficiency, I'll go with this option.